lib/smart_answer/calculators/arrested_abroad.rb

```diff
@@ -1,14 +1,12 @@
 module SmartAnswer::Calculators
   class ArrestedAbroad
     # created for the help-if-you-are-arrested-abroad calculator
-    attr_reader :data
+    attr_accessor :country
 
-    def initialize
-      @data = self.class.prisoner_packs
-    end
+    PRISONER_PACKS = YAML.load_file(Rails.root.join("config/smart_answers/prisoner_packs.yml")).freeze
 
     def generate_url_for_download(country, field, text)
```

Comment:
Why is country being passed in when it's an attribute of the instance?

This class could be simplified if it was written as an object and not a collection of functional snippets.

Reply:
This method should be rewritten to remove the use of single letter variables on multi-line code, and simply the code:
```ruby
def country_data
  @country_data ||= PRISONER_PACKS.find { |c| c["slug"] == country }
end

def generate_url_for_download(field, text)
  return "" unless country_data && country_data[field]

  urls = country_data[field].split(" ")
  output = urls.map do |url|
    new_link = "- [#{text}](#{url})"
    new_link += '{:rel="external"}' if url.include? "http"
  end
  output.join("\n")
end
```